src/com/zone5cloud/retrofit/core/apis/ThirdPartyTokenAPI.java

```diff
@@ -48,4 +46,38 @@
     @DELETE(Users.DEREGISTER_DEVICE_THIRD_PARTY_CONNECTION)
     Observable<Response<Void>> deregisterDeviceWithThirdParty(@Path("token") String token);
 
+    /**
```

Comment:
Can you deprecate the 3 endpoints that you are replacing? setThirdPartyToken, removeThirdPartyToken and hasThirdyPartyToken?